modules/tensor_mechanics/src/materials/ADComputeIsotropicElasticityTensorShell.C

```diff
@@ -0,0 +1,88 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ADComputeIsotropicElasticityTensorShell.h"
+#include "RankFourTensor.h"
+
+#include "libmesh/quadrature.h"
+#include "libmesh/utility.h"
+#include "libmesh/enum_quadrature_type.h"
+#include "libmesh/fe_type.h"
+#include "libmesh/string_to_enum.h"
+#include "libmesh/quadrature_gauss.h"
+
+registerADMooseObject("TensorMechanicsApp", ADComputeIsotropicElasticityTensorShell);
+
+defineADValidParams(
+    ADComputeIsotropicElasticityTensorShell,
+    ADMaterial,
+    params.addClassDescription("Compute a plane stress isotropic elasticity tensor.");
```

Comment:
Just a comment -- we could probably use this for plane stress elasticity problems as well, except the fact that we have the through-thickness index requires a little more work.